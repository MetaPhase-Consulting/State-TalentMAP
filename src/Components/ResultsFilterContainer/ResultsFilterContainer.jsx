import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FILTER_ITEMS_ARRAY, ACCORDION_SELECTION_OBJECT,
  MISSION_DETAILS_ARRAY, POST_DETAILS_ARRAY } from '../../Constants/PropTypes';
import { ACCORDION_SELECTION } from '../../Constants/DefaultProps';
import SearchFiltersContainer from '../SearchFilters/SearchFiltersContainer/SearchFiltersContainer';
import ResetFilters from '../ResetFilters/ResetFilters';

class ResultsFilterContainer extends Component {
  shouldComponentUpdate(nextProps) {
    if (this.props !== nextProps) {
      return true;
    }
    return false;
  }

  render() {
    const { filters, resetFilters, setAccordion, selectedAccordion,
      fetchMissionAutocomplete, missionSearchResults, missionSearchIsLoading,
      missionSearchHasErrored, fetchPostAutocomplete, onQueryParamUpdate, onQueryParamToggle,
      postSearchResults, postSearchIsLoading, postSearchHasErrored } = this.props;
    return (
      <div className="filter-container">
        <div className="filter-container-bottom">
          <div className="usa-grid-full filter-control-container">
            <div className="filter-control-left">Select Filter:</div>
            <div className="filter-control-right">
              <ResetFilters resetFilters={resetFilters} />
            </div>
          </div>
          <div className="usa-grid-full search-filters-container">
            <SearchFiltersContainer
              filters={filters}
              queryParamUpdate={onQueryParamUpdate}
              queryParamToggle={onQueryParamToggle}
              selectedAccordion={selectedAccordion}
              setAccordion={setAccordion}
              fetchMissionAutocomplete={fetchMissionAutocomplete}
              missionSearchResults={missionSearchResults}
              missionSearchIsLoading={missionSearchIsLoading}
              missionSearchHasErrored={missionSearchHasErrored}
              fetchPostAutocomplete={fetchPostAutocomplete}
              postSearchResults={postSearchResults}
              postSearchIsLoading={postSearchIsLoading}
              postSearchHasErrored={postSearchHasErrored}
            />
          </div>
        </div>
      </div>
    );
  }
}

ResultsFilterContainer.propTypes = {
  filters: FILTER_ITEMS_ARRAY.isRequired,
  onQueryParamUpdate: PropTypes.func.isRequired,
  onQueryParamToggle: PropTypes.func.isRequired,
  resetFilters: PropTypes.func.isRequired,
  setAccordion: PropTypes.func.isRequired,
  selectedAccordion: ACCORDION_SELECTION_OBJECT,
  fetchMissionAutocomplete: PropTypes.func.isRequired,
  missionSearchResults: MISSION_DETAILS_ARRAY.isRequired,
  missionSearchIsLoading: PropTypes.bool.isRequired,
  missionSearchHasErrored: PropTypes.bool.isRequired,
  fetchPostAutocomplete: PropTypes.func.isRequired,
  postSearchResults: POST_DETAILS_ARRAY.isRequired,
  postSearchIsLoading: PropTypes.bool.isRequired,
  postSearchHasErrored: PropTypes.bool.isRequired,
};

ResultsFilterContainer.defaultProps = {
  selectedAccordion: ACCORDION_SELECTION,
};

export default ResultsFilterContainer;
