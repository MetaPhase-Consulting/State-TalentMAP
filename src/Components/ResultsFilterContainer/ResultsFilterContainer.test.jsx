import { mount, shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
// import { setupAsyncMocks } from '../testUtilities/testUtilities';
import ResultsFilterContainer from './ResultsFilterContainer';
import { bidderUserObject } from '../../__mocks__/userObject';

describe('ResultsFilterContainerComponent', () => {
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  // const { mockStore } = setupAsyncMocks();
  // let [store] = Array(1);

  // store = mockStore({ clientView: {} });


  const items = [{
    title: 'title', expanded: true,
  }];

  const props = {
    filters: items,
    onQueryParamUpdate: () => {},
    onQueryParamToggle: () => {},
    resetFilters: () => {},
    setAccordion: () => {},
    fetchMissionAutocomplete: () => {},
    missionSearchResults: [],
    missionSearchIsLoading: false,
    missionSearchHasErrored: false,
    fetchPostAutocomplete: () => {},
    postSearchResults: [],
    postSearchIsLoading: false,
    postSearchHasErrored: false,
    userProfile: bidderUserObject,
    isLoading: false,
    showClear: true,
    clientView: {},
    client: {},
  };

  it('is defined', () => {
    const wrapper = shallow(
      <Provider store={mockStore({})}>
        <MemoryRouter>
          <ResultsFilterContainer
            {...props}
          />
        </MemoryRouter>
      </Provider>,
      );
    expect(wrapper).toBeDefined();
  });

  it('can receive props', () => {
    const wrapper = mount(
      <Provider store={mockStore({ clientView: {} })}>
        <MemoryRouter>
          <ResultsFilterContainer
            {...props}
          />
        </MemoryRouter>
      </Provider>,
      );
    console.log('===');
    console.log(wrapper.instance().props);
    console.log('===');
    expect(wrapper.instance().props.filters).toBe(items);
  });

  it('is defined after receiving new props', () => {
    const wrapper = shallow(
      <Provider store={mockStore({})}>
        <MemoryRouter>
          <ResultsFilterContainer
            {...props}
          />
        </MemoryRouter>
      </Provider>,
      );
    wrapper.setProps({ newProp: 'abz' });
    expect(wrapper).toBeDefined();
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <Provider store={mockStore({})}>
        <MemoryRouter>
          <ResultsFilterContainer
            {...props}
          />
        </MemoryRouter>
      </Provider>,
      );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot when isLoading === true', () => {
    const wrapper = shallow(
      <Provider store={mockStore({})}>
        <MemoryRouter>
          <ResultsFilterContainer
            {...props}
            isLoading
          />
        </MemoryRouter>
      </Provider>,
      );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
