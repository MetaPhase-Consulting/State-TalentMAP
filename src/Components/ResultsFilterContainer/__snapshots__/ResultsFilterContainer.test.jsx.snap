// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsFilterContainerComponent matches snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <MemoryRouter>
    <ResultsFilterContainer
      client={Object {}}
      clientView={Object {}}
      fetchMissionAutocomplete={[Function]}
      fetchPostAutocomplete={[Function]}
      filters={
        Array [
          Object {
            "expanded": true,
            "title": "title",
          },
        ]
      }
      isLoading={false}
      missionSearchHasErrored={false}
      missionSearchIsLoading={false}
      missionSearchResults={Array []}
      onQueryParamToggle={[Function]}
      onQueryParamUpdate={[Function]}
      postSearchHasErrored={false}
      postSearchIsLoading={false}
      postSearchResults={Array []}
      resetFilters={[Function]}
      selectedAccordion={
        Object {
          "main": "",
          "sub": "",
        }
      }
      setAccordion={[Function]}
      showClear={true}
      userProfile={
        Object {
          "bid_statistics": Array [
            Object {
              "draft": 1,
              "submitted": 1,
            },
          ],
          "cdo": Object {
            "email": "shadtrachl@state.gov",
            "first_name": "Leah",
            "initials": "LS",
            "last_name": "Shadtrach",
            "username": "shadtrachl",
          },
          "display_name": "John",
          "emp_id": "1",
          "employee_info": Object {
            "grade": "00",
            "skills": Array [
              Object {
                "code": "6218",
                "description": "CONSTRUCTION ENGINEERING",
              },
              Object {
                "code": "6211",
                "description": "COMPUTER SCIENCE",
              },
            ],
          },
          "employee_profile_url": Object {
            "external": "https://external.com",
            "internal": "https://internal.com",
          },
          "favorite_positions": Array [
            Object {
              "id": 1,
              "representation": "[00003026] OMS (COM) (Freetown, Sierra Leone)",
            },
            Object {
              "id": 4,
              "representation": "[00180000] OMS (DCM) (Addis Ababa, Ethiopia)",
            },
          ],
          "favorite_positions_pv": Array [
            Object {
              "id": 10,
            },
            Object {
              "id": 20,
            },
          ],
          "favorite_tandem_positions": Array [
            Object {
              "id": 1,
              "representation": "[00003026] OMS (COM) (Freetown, Sierra Leone)",
            },
            Object {
              "id": 4,
              "representation": "[00180000] OMS (DCM) (Addis Ababa, Ethiopia)",
            },
          ],
          "favorite_tandem_positions_pv": Array [
            Object {
              "id": 10,
            },
            Object {
              "id": 20,
            },
          ],
          "grade": "03",
          "id": 1,
          "initials": "JD",
          "is_cdo": false,
          "skills": Array [
            63,
            40,
          ],
          "user": Object {
            "email": "doej@state.gov",
            "first_name": "John",
            "initials": "JD",
            "last_name": "Doe",
            "username": "jdoe",
          },
          "user_info": Object {
            "office_address": "1234 Apple St. NW Washington, DC 20010",
            "office_phone": "202-876-5432",
          },
        }
      }
    />
  </MemoryRouter>
</ContextProvider>
`;

exports[`ResultsFilterContainerComponent matches snapshot when isLoading === true 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <MemoryRouter>
    <ResultsFilterContainer
      client={Object {}}
      clientView={Object {}}
      fetchMissionAutocomplete={[Function]}
      fetchPostAutocomplete={[Function]}
      filters={
        Array [
          Object {
            "expanded": true,
            "title": "title",
          },
        ]
      }
      isLoading={true}
      missionSearchHasErrored={false}
      missionSearchIsLoading={false}
      missionSearchResults={Array []}
      onQueryParamToggle={[Function]}
      onQueryParamUpdate={[Function]}
      postSearchHasErrored={false}
      postSearchIsLoading={false}
      postSearchResults={Array []}
      resetFilters={[Function]}
      selectedAccordion={
        Object {
          "main": "",
          "sub": "",
        }
      }
      setAccordion={[Function]}
      showClear={true}
      userProfile={
        Object {
          "bid_statistics": Array [
            Object {
              "draft": 1,
              "submitted": 1,
            },
          ],
          "cdo": Object {
            "email": "shadtrachl@state.gov",
            "first_name": "Leah",
            "initials": "LS",
            "last_name": "Shadtrach",
            "username": "shadtrachl",
          },
          "display_name": "John",
          "emp_id": "1",
          "employee_info": Object {
            "grade": "00",
            "skills": Array [
              Object {
                "code": "6218",
                "description": "CONSTRUCTION ENGINEERING",
              },
              Object {
                "code": "6211",
                "description": "COMPUTER SCIENCE",
              },
            ],
          },
          "employee_profile_url": Object {
            "external": "https://external.com",
            "internal": "https://internal.com",
          },
          "favorite_positions": Array [
            Object {
              "id": 1,
              "representation": "[00003026] OMS (COM) (Freetown, Sierra Leone)",
            },
            Object {
              "id": 4,
              "representation": "[00180000] OMS (DCM) (Addis Ababa, Ethiopia)",
            },
          ],
          "favorite_positions_pv": Array [
            Object {
              "id": 10,
            },
            Object {
              "id": 20,
            },
          ],
          "favorite_tandem_positions": Array [
            Object {
              "id": 1,
              "representation": "[00003026] OMS (COM) (Freetown, Sierra Leone)",
            },
            Object {
              "id": 4,
              "representation": "[00180000] OMS (DCM) (Addis Ababa, Ethiopia)",
            },
          ],
          "favorite_tandem_positions_pv": Array [
            Object {
              "id": 10,
            },
            Object {
              "id": 20,
            },
          ],
          "grade": "03",
          "id": 1,
          "initials": "JD",
          "is_cdo": false,
          "skills": Array [
            63,
            40,
          ],
          "user": Object {
            "email": "doej@state.gov",
            "first_name": "John",
            "initials": "JD",
            "last_name": "Doe",
            "username": "jdoe",
          },
          "user_info": Object {
            "office_address": "1234 Apple St. NW Washington, DC 20010",
            "office_phone": "202-876-5432",
          },
        }
      }
    />
  </MemoryRouter>
</ContextProvider>
`;
